std/hash/poseidon_gl.asm

```diff
@@ -57,26 +51,28 @@ machine PoseidonGL(LASTBLOCK, operation_id) {
     pol constant C_11 = [0xc54302f225db2c76, 0xc0572c8c08cbbbad, 0x28b8ec3ae9c29633, 0xa637093ecb2ad631, 0xfac6ff1346a41675, 0xb65481ba645c602, 0x6604df4fee32bcb1, 0xa6217d8bf660f29c, 0x5d693c1ba3ba3621, 0xb0a7eae879ddb76d, 0x28fd3b046380f850, 0x1c4dbb1c4c27d243, 0x25d64362697828fd, 0xdd5118375bf1a9b9, 0x64bb6dec369d4418, 0x8ad97b33f1ac1455, 0x4f0a2e5fb028f2ce, 0x6d1f5a59005bec17, 0xeeb76e397069e804, 0x9ef1cd60e679f284, 0xdfe89923f6c9c2ff, 0x9001a64209f21db8, 0x9e8cd55b57637ed0, 0x7248fe7705f38e47, 0x4a20358574454ec0, 0x29bff3ecb9b7a6e3, 0xc02ac5e47312d3ca, 0xa49229d24d014343, 0x164bb2de1bbeddc8, 0xbc8dfb627fe558fc, 0x0]*;
     
     // State of the Poseidon permutation
-    pol commit in0, in1, in2, in3, in4, in5, in6, in7, cap0, cap1, cap2, cap3;
+    pol commit inp[8];
+    pol commit cap[4];
 
     // The initial state of the Poseidon permutation
-    // (constrained to be equal to (in0, in1, in2, in3, in4, in5, in6, in7, cap0, cap1, cap2, cap3)
+    // (constrained to be equal to (inp[0], inp[1], ..., inp[7], cap[0], cap[1], cap[2], cap[3])
     // in the first row and then repeated until the end of the block)
-    pol commit input_in0, input_in1, input_in2, input_in3, input_in4, input_in5, input_in6, input_in7, input_cap0, input_cap1, input_cap2, input_cap3;
+    pol commit input_inp[8];
+    pol commit input_cap[4];
 
     // Add round constants
-    pol a0 = in0 + C_0;
-    pol a1 = in1 + C_1;
-    pol a2 = in2 + C_2;
-    pol a3 = in3 + C_3;
-    pol a4 = in4 + C_4;
-    pol a5 = in5 + C_5;
-    pol a6 = in6 + C_6;
-    pol a7 = in7 + C_7;
-    pol a8 = cap0 + C_8;
-    pol a9 = cap1 + C_9;
-    pol a10 = cap2 + C_10;
-    pol a11 = cap3 + C_11;
+    pol a0 = inp[0] + C_0;
+    pol a1 = inp[1] + C_1;
+    pol a2 = inp[2] + C_2;
+    pol a3 = inp[3] + C_3;
+    pol a4 = inp[4] + C_4;
+    pol a5 = inp[5] + C_5;
+    pol a6 = inp[6] + C_6;
+    pol a7 = inp[7] + C_7;
+    pol a8 = cap[0] + C_8;
+    pol a9 = cap[1] + C_9;
+    pol a10 = cap[2] + C_10;
+    pol a11 = cap[3] + C_11;
```

Comment:
I guess all these blocks here could also be transformed into folds?

Reply:
The ones below, yes, but I'm not sure about intermediate polys.